.travis.yml

```diff
@@ -36,6 +36,10 @@ matrix:
       env: DJANGO="django>=1.10.0,<1.11.0" REST_FRAMEWORK="djangorestframework>=3.4,<3.5"
     - python: "3.3"
       env: DJANGO="django>=1.9.0,<1.10.0" REST_FRAMEWORK="djangorestframework>=3.4,<3.5"
+  include:
+    - python: "3.5"
+      env: W='python -W error -m' DJANGO='https://github.com/django/django/archive/master.tar.gz' REST_FRAMEWORK='djangorestframework>=3.4,<3.5'
   allow_failures:
+    - env: W='python -W error -m' DJANGO='https://github.com/django/django/archive/master.tar.gz' REST_FRAMEWORK='djangorestframework>=3.4,<3.5'
```

Comment:
You should be able to set the `-W error` flag using the [`PYTHONWARNINGS` environment variable](http://stackoverflow.com/q/14463277/359284#17654868). This way you wouldn't need to set a custom `W` environment variable.

https://docs.python.org/3.4/using/cmdline.html#envvar-PYTHONWARNINGS